src/Elastic.Configuration/FileBased/Yaml/YamlSettingsBase.cs

```diff
@@ -21,10 +22,30 @@ protected YamlConfigurationBase(string path, IFileSystem fileSystem)
 
 			try
 			{
-				var raw = this.FileSystem.File.ReadAllText(Path);
+				var rawText = this.FileSystem.File.ReadAllText(Path);
+
+				// we flatten because neither yamldotnet nor sharpyaml can bind nested.props in all variants
+				var rawYaml = this._serializer.Deserialize<Dictionary<string, object>>(rawText);
+				var dict = Flatten(rawYaml);
+
+				// these might be lists disguised as strings, fix them
+				var maybeLists = new[]
+				{
+					"discovery.seed_hosts",
+					"cluster.initial_master_nodes"
+				};
+
+				foreach (var item in maybeLists)
+				{
+					if (dict[item] is string value)
```

Comment:
should this be more defensive and use `TryGetValue()`? For example, `cluster.initial_master_nodes` only needs to be specified on the initial cluster bootstrap as far as I am aware, so it is _possible_ to not exist in the configuration on upgrade for nodes that were not part of the initial bootstrap.

Reply:
Indeed .. thanks for catching that! Will fix.